eclair-core/src/test/scala/fr/acinq/eclair/router/GraphSpec.scala

```diff
@@ -239,4 +238,19 @@ class GraphSpec extends AnyFunSuite {
 
   def edgeFromNodes(shortChannelId: Long, a: PublicKey, b: PublicKey): GraphEdge = makeEdge(shortChannelId, a, b, 0 msat, 0)
 
+  test("amount with fees larger than channel capacity") {
```

Comment:
Please add a diagram, it's much easier to understand this test and verify that the route with the problematic channel capacity is taken into account.

There are a few things I don't understand about this test.
First, why is the D -> C channel needed?
Second, the faulty route (C -> D -> E) costs more fees than the route C -> E, so it's really not obvious that it's considered by the algorithm. I think you should make that route cheaper than C -> E to help readers see that it's the best route, but it has a capacity issue.

Last, I think we should run  `yenKshortestPaths` with both `HeuristicsConstants` and `WeightRatios`, it's good to test both of them in case we introduce a similar issue in the `WeightRatios` case in the future (we would catch it instantly).

Reply:
Should I just remove it then?

Reply:
No, clearly not, but it should contain comments to explain what the issue was and how it's tested. Just imagine someone reading this test in one year, and think about how you could help them understand what this is testing and why it's important.